processor/internal/transformer/destination_transformer/embedded/kafka/kafka_test.go

```diff
@@ -0,0 +1,909 @@
+						"messageId": "message-id-4",
+					},
+					Destination: backendconfig.DestinationT{
+						ID:                    destinationWithNoConfigTopic.ID,
+						DestinationDefinition: destinationWithNoConfigTopic.DestinationDefinition,
+						WorkspaceID:           destinationWithNoConfigTopic.WorkspaceID,
+						Config: map[string]interface{}{
+							"topic": "",
+						},
+					},
+					Metadata: metadataWithRudderID,
+				},
+			},
+			want: types.Response{
+				FailedEvents: []types.TransformerResponse{
+					{
+						Error:      "topic is required for Kafka destination",
+						Metadata:   metadataWithRudderID,
+						StatusCode: http.StatusInternalServerError,
+						StatTags:   destinationWithNoConfigTopicStatTags,
+					},
+					{
+						Error:      "topic is required for Kafka destination",
+						Metadata:   metadataWithRudderID,
+						StatusCode: http.StatusInternalServerError,
+						StatTags:   destinationWithNoConfigTopicStatTags,
+					},
+					{
+						Error:      "topic is required for Kafka destination",
+						Metadata:   metadataWithRudderID,
+						StatusCode: http.StatusInternalServerError,
+						StatTags:   destinationWithNoConfigTopicStatTags,
+					},
+				},
+				Events: []types.TransformerResponse{
+					{
+						Output: map[string]interface{}{
+							"message": map[string]interface{}{
+								"userId":    "user-123",
+								"messageId": "message-id-3",
+								"integrations": map[string]interface{}{
+									"kafka": map[string]interface{}{
+										"topic": "default-topic",
+									},
+								},
+							},
+							"topic":  "default-topic",
+							"userId": "user-123",
+						},
+						Metadata:   expectedMetadataWithDefaultTopic,
+						StatusCode: http.StatusOK,
+					},
+				},
+			},
+		},
+		{
+			name: "should set correct timestamp for retl event",
+			events: []types.TransformerEvent{
+				{
+					Message: map[string]interface{}{
+						"userId":  "user-123",
+						"type":    "identify",
+						"channel": "sources",
+						"context": map[string]interface{}{
+							"timestamp": "2021-01-01T00:00:00Z",
+						},
+					},
+					Destination: destinationWithConfigTopic,
+					Metadata:    metadataWithRudderID,
+				},
+			},
+			want: types.Response{
+				Events: []types.TransformerResponse{
+					{
+						Output: map[string]interface{}{
+							"message": map[string]interface{}{
+								"userId":    "user-123",
+								"type":      "identify",
+								"channel":   "sources",
+								"timestamp": "2021-01-01T00:00:00Z",
+								"context": map[string]interface{}{
+									"timestamp": "2021-01-01T00:00:00Z",
+								},
+							},
+							"topic":  "default-topic",
+							"userId": "user-123",
+						},
+						Metadata:   expectedMetadataWithDefaultTopic,
+						StatusCode: http.StatusOK,
+					},
+				},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			got := Transform(context.Background(), c.events)
+			require.Equal(t, c.want, got)
+		})
+	}
+}
+
```

Comment:
This [comment](https://github.com/rudderlabs/rudder-server/pull/5525#discussion_r2028574533), applies here as well. Can be done for both after merging